src/path_buffer.rs

```diff
@@ -24,6 +24,10 @@ impl PathBuffer {
     pub fn join<P: AsRef<Path>>(&self, path: P) -> Self {
         Self::from(self.0.join(path))
     }
+
+    pub fn maybe_from(value: Option<&str>) -> Option<Self> {
```

Comment:
In substrate I've observed that they use TryFrom and TryInto to return Result, so it may fail.
Here you use _maybe_ it with an option, is that something standard? I'm not familiar with it.